youtube_dl/extractor/canvas.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class CanvasIE(InfoExtractor):
+    _VALID_URL = r'(?:https?://)?(?:www\.)?canvas\.be/video/(?P<id>((([[a-z0-9]+-?)+/?)+))'
+    _TEST = {
+        'url': 'http://www.canvas.be/video/radio-gaga/najaar-2015/internaat-turnhout',
+        'md5': '11ea0075408f081bef2278b7ebeb560b',
+        'info_dict': {
+            'id': 'radio-gaga/najaar-2015/internaat-turnhout',
+            'title': 'Internaat Turnhout',
+            'ext': 'mp4',
+            'duration': 2467160,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(
+            r'<h1 class="video__body__header__title">(.*?)</h1>', webpage, 'title')
+        data_video = self._html_search_regex(
+            r'data-video="(.*?)"', webpage, 'data-video')
+        json_url = 'https://mediazone.vrt.be/api/v1/canvas/assets/' + data_video
+        data = self._download_json(json_url, video_id)
+
+        formats = []
+        for target in data['targetUrls']:
+            if target['type'] == 'PROGRESSIVE_DOWNLOAD':
```

Comment:
Missing `type` and `url` in `target` should be tolerated without extractor breakage.